src/frontend/src/utils/redact.rs

```diff
@@ -0,0 +1,646 @@
+        Err(Error::NotSupported("serialize_unit".into()))
+    }
+
+    fn serialize_unit_struct(self, _name: &'static str) -> Result<Self::Ok, Self::Error> {
+        Err(Error::NotSupported("serialize_unit_struct".into()))
+    }
+
+    fn serialize_unit_variant(
+        self,
+        _name: &'static str,
+        _variant_index: u32,
+        _variant: &'static str,
+    ) -> Result<Self::Ok, Self::Error> {
+        Err(Error::NotSupported("serialize_unit_variant".into()))
+    }
+
+    fn serialize_newtype_struct<T: ?Sized>(
+        self,
+        _name: &'static str,
+        value: &T,
+    ) -> Result<Self::Ok, Self::Error>
+    where
+        T: Serialize,
+    {
+        value.serialize(self)
+    }
+
+    fn serialize_newtype_variant<T: ?Sized>(
+        self,
+        _name: &'static str,
+        _variant_index: u32,
+        _variant: &'static str,
+        value: &T,
+    ) -> Result<Self::Ok, Self::Error>
+    where
+        T: Serialize,
+    {
+        value.serialize(self)
+    }
+
+    fn serialize_seq(self, _len: Option<usize>) -> Result<Self::SerializeSeq, Self::Error> {
+        Err(Error::NotSupported("serialize_seq".into()))
+    }
+
+    fn serialize_tuple(self, _len: usize) -> Result<Self::SerializeTuple, Self::Error> {
+        Err(Error::NotSupported("serialize_tuple".into()))
+    }
+
+    fn serialize_tuple_struct(
+        self,
+        _name: &'static str,
+        _len: usize,
+    ) -> Result<Self::SerializeTupleStruct, Self::Error> {
+        Err(Error::NotSupported("serialize_tuple_struct".into()))
+    }
+
+    fn serialize_tuple_variant(
+        self,
+        _name: &'static str,
+        _variant_index: u32,
+        _variant: &'static str,
+        _len: usize,
+    ) -> Result<Self::SerializeTupleVariant, Self::Error> {
+        Err(Error::NotSupported("serialize_tuple_variant".into()))
+    }
+
+    fn serialize_map(self, _len: Option<usize>) -> Result<Self::SerializeMap, Self::Error> {
+        Ok(self)
+    }
+
+    fn serialize_struct(
+        self,
+        _name: &'static str,
+        _len: usize,
+    ) -> Result<Self::SerializeStruct, Self::Error> {
+        Ok(self)
+    }
+
+    fn serialize_struct_variant(
+        self,
+        _name: &'static str,
+        _variant_index: u32,
+        _variant: &'static str,
+        _len: usize,
+    ) -> Result<Self::SerializeStructVariant, Self::Error> {
+        Err(Error::NotSupported("serialize_struct_variant".into()))
+    }
+}
+
+pub fn try_redact_definition(definition: &str) -> Result<String, Error> {
+    use itertools::Itertools;
+    let ast = risingwave_sqlparser::parser::Parser::parse_sql(definition)
+        .map_err(|e| Error::Internal(e.into()))?;
+    let stmt = ast
+        .into_iter()
+        .exactly_one()
+        .map_err(|e| Error::Internal(e.into()))?;
+    try_redact_statement(&stmt).map(|stmt| stmt.to_string())
+}
+
+fn try_redact_statement(stmt: &Statement) -> Result<Statement, Error> {
```

Comment:
Let me try to summarize my understanding of this PR: 
- `try_redact_statement` (`Statement` -> `Statement`) is the end goal we want to achieve.
- The implementation is `Statement` -> `ConnectorProperties` -> `Statement`.
	- The `ConnectorProperties` -> `Statement` is achieved by the `SqlOptionVecSerializer`, and the `Serialize` impl (which actually does the redaction)

This approach generally sounds good to me. But I have some thoughts:
- In order to redact `Statement` -> `Statement`, an easier approach IMO is just modifying the hashmap directly. We can check the key's name (like having a blacklist for redaction). 
- The redaction needs to be performed manually, so we may still access original SQL in e.g., SQL log. I don't know if this is good enough.

Reply:
Oh, https://github.com/risingwavelabs/risingwave/pull/14310 looks very like the solution in my mind, but it contains more stuff, including redacting table/column names. Why don't we go with that approach? 🤔

Reply:
> In order to redact Statement -> Statement, an easier approach IMO is just modifying the hashmap directly. We can check the key's name (like having a blacklist for redaction).

Yes, #14310 basically redact all with_options values, i.e. an empty whitelist.

Reply:
The upside of this PR is it's strong typed, and secrets in ConnectorProperties won't be printed unexpectedly.

Reply:
> secrets in ConnectorProperties won't be printed unexpectedly

I agree this is quite a good change. 

However, previously `ConnectorProperties` doesn't implements `Serialize` (and `Debug`) (Actually I removed some unnecessary `#[derive(Serialize)]`), so probably it won't be printed unexpectedly. 

Now we adds `#[derive(Serialize)]` (although redacted), I feel this isn't very cool..

Reply:
> However, previously ConnectorProperties doesn't implements Serialize (and Debug)

It seems I'm wrong. It already implemented `Debug`. 🤡